druntime/src/core/builtins.d

```diff
@@ -51,3 +51,32 @@ version (LDC)
 
 /// Writes `s` to `stderr` during CTFE (does nothing at runtime).
 void __ctfeWrite(scope const(char)[] s) @nogc @safe pure nothrow {}
+
+version (GNU)
+{
+    alias expect = __builtin_expect;
+    alias trap = __builtin_trap;
+}
+else version (LDC)
+{
+    alias expect = llvm_expect;
+    debug
+        alias trap = llvm_debugtrap;
+    else
+        alias trap = llvm_trap;
+}
+else version (DigitalMars)
+{
+    pragma(inline, true)
+    T expect(T)(T val, T expected) if (__traits(isIntegral, T))
+    {
+        return val;
+    }
+
+    pragma(inline, true)
+    noreturn trap() { assert(0); }
```

Comment:
Can debug trap be implemented like this?
```suggestion
    void trap() nothrow @nogc pure @safe
    {
        debug
        {
            version(D_InlineAsm_X86_64) 
		        asm nothrow @nogc pure @trusted {int 3;}
		    else
		        assert(0); // debug trap not implemented
	    }
	    else
            assert(0); 
```

Reply:
int 3 works in 32 bit x64 asm too.

Reply:
Done